cmd/parse-genesis/utils/utils.go

```diff
@@ -0,0 +1,105 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	parsecmd "github.com/forbole/juno/v2/cmd/parse"
+	"github.com/forbole/juno/v2/database"
+	"github.com/forbole/juno/v2/modules"
+	modsregistrar "github.com/forbole/juno/v2/modules/registrar"
+	"github.com/forbole/juno/v2/types/config"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// GetRegisteredModules returns the registered modules from the config file
+func GetRegisteredModules(parseConfig *parsecmd.Config) ([]modules.Module, error) {
+	// Get the global config
+	cfg := config.Cfg
+
+	// Build the codec
+	encodingConfig := parseConfig.GetEncodingConfigBuilder()()
+
+	// Setup the SDK configuration
+	sdkConfig := sdk.GetConfig()
+	parseConfig.GetSetupConfig()(cfg, sdkConfig)
+	sdkConfig.Seal()
+
+	// Get the db
+	databaseCtx := database.NewContext(cfg.Database, &encodingConfig, parseConfig.GetLogger())
+	db, err := parseConfig.GetDBBuilder()(databaseCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the modules
+	context := modsregistrar.NewContext(cfg, sdkConfig, &encodingConfig, db, nil, parseConfig.GetLogger())
+	mods := parseConfig.GetRegistrar().BuildModules(context)
+	registeredModules := modsregistrar.GetModules(mods, cfg.Chain.Modules, parseConfig.GetLogger())
+
+	return registeredModules, nil
+}
+
+// GetGenesisDocAndState generates and returns the genesis doc and genesis state with genesis.json file
+func GetGenesisDocAndState() (*tmtypes.GenesisDoc, map[string]json.RawMessage, error) {
```

Comment:
Currently this function is completely ignoring the fact that the user might have selected to not use a genesis file but rather fetch the genesis directly from the node instead. For this reason, the overall behavior of `GetGenesisDocAndState` should be the following: 

1. Look at the config `GenesisFilePath` value specified. 
2. If the path is **empty**, fetch the genesis doc from the node. 
3. If the path is **not empty**, read the genesis from the file. 
4. In both cases, get the genesis state from the genesis doc

I also think that since this method might be useful elsewhere, it might be better to implement it inside the Juno library instead of here. Currently Juno uses the following code to get the genesis doc: 

```go
cfg := config.Cfg.Parser

var genesis *tmtypes.GenesisDoc
if strings.TrimSpace(cfg.GenesisFilePath) != "" {
    genesis, err = w.getGenesisFromFilePath(cfg.GenesisFilePath)
    if err != nil {
        return err
    }
} else {
    genesis, err = w.getGenesisFromRPC()
    if err != nil {
        return err
    }
}


// ---

// getGenesisFromRPC returns the genesis read from the RPCConfig endpoint
func (w Worker) getGenesisFromRPC() (*tmtypes.GenesisDoc, error) {
    w.logger.Debug("getting genesis from RPCConfig")

    response, err := w.node.Genesis()
    if err != nil {
        return nil, fmt.Errorf("failed to get genesis: %s", err)
    }

    return response.Genesis, nil
}

// getGenesisFromFilePath tries reading the genesis doc from the given path
func (w Worker) getGenesisFromFilePath(path string) (*tmtypes.GenesisDoc, error) {
    w.logger.Debug("reading genesis from file", "path", path)

    bz, err := tmos.ReadFile(path)
    if err != nil {
        return nil, fmt.Errorf("failed to read genesis file: %s", err)
    }

    var genDoc tmtypes.GenesisDoc
    err = tmjson.Unmarshal(bz, &genDoc)
    if err != nil {
        return nil, fmt.Errorf("failed to unmarshal genesis doc: %s", err)
    }

    return &genDoc, nil
}
```

Instead I think we can make it more general, something like this: 
```go
func GetGenesisDocAndState(parserCfg parserconfig.Config, node node.Node) (*tmtypes.GenesisDoc, map[string[json.RawMessage], error) {
    ...
}
```

This way this function can then be used also by BDJuno as well as other libraries using Juno. 
```